File: src/java/pointingpoker/model/Member.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package pointingpoker.model;


public class Member {

    private int id;
    private String name;
//    private String roomKey = "";
    private String vote = "";
    private Integer clientKey = 0;
    private Boolean observer = false;

    public Member() {
    }
    
    public int getId() {
        return id;
    }
    
    public Integer getClientKey() {
        return clientKey;
    }

    public String getName() {
        return name;
    }

//	public String getRoomKey() {
//        return roomKey;
//    }

    public String getVote() {
        return vote;
    }

    public Boolean getObserver() {
        return observer;
    }

    public void setId(int id) {
        this.id = id;
    }
    
    public void setClientKey(Integer clientKey) {
        this.clientKey = clientKey;
    }

    public void setName(String name) {
        this.name = name;
    }

//    public void setRoomKey(String roomKey) {
//        this.roomKey = roomKey;
//    }

    public void setVote(String vote) {
        this.vote = vote;
    }
    
    public void setObserver(Boolean observer) {
        this.observer = observer;
    }
}

File: app/models/Member.js

function Member(_req){

  var _member = {
    id: 0,
    name: "",
    observer: "",
    vote: "",
    clientKey: 0,
    roomKey: ""
  };

  _member.id = parseInt(Math.random()*10000000,10);

  for(var _val in _member){
    if(typeof(_req[_val]) === "undefined"){
      _req[_val] = _member[_val];
    }
  }

  return _req;

}

module.exports = Member;
